### PR TITLE
Added MultiDayView to RadCalendar

### PR DESCRIPTION
**Pull request by APopatanasov**
- Created MultiDay view for the Calendar;
- Add all day area for the appointments for the MultiDayView;
- Customization for TimeSlot;
- Add Tap command for the TimeSlot and Appointment;
- Customization for the allday area;
- Add ScrollToAppointment API;
- Introduce swiping for the Week and Day area for touch;
- Current time indicator;
- Navigation with the mouse for desktop;

### CODE REVIEW

File: Controls/Input/Input.UWP/Calendar/View/Services/Commands/CommandId.cs [hunk 1]
```diff
@@ -43,6 +43,16 @@ public enum CommandId
...
+
+        /// <summary>
+        /// A command associated with the Tap event that occurs on tapping an appointment.
+        /// </summary>
+        AppointmentTap,
+
+        /// <summary>
+        /// A command associated with the Tap event that occurs on tapping a time slot.
+        /// </summary>
+        TimeSlotTap
```
Comment (ni4ka7a) [inline Controls/Input/Input.UWP/Calendar/View/Services/Commands/CommandId.cs]: 🥇

File: Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs [hunk 2]
```diff
@@ -264,7 +273,38 @@ public partial class RadCalendar : RadControl, IView, ICultureAware
...
+            DependencyProperty.Register(nameof(HeaderVisibility), typeof(Visibility), typeof(RadCalendar), new PropertyMetadata(Visibility.Visible));
+
+        /// <summary>
+        /// Identifies the <c cref="NavigationControlBorderStyle"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty NavigationControlBorderStyleProperty =
+            DependencyProperty.Register(nameof(NavigationControlBorderStyle), typeof(Style), typeof(RadCalendar), new PropertyMetadata(null));
+
         internal static readonly Size InfinitySize = new Size(double.PositiveInfinity, double.PositiveInfinity);
+        internal static ResourceDictionary EmbeddedResourceDict;
```
Comment (ni4ka7a) [inline Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs]: Rename to something like MultiDayViewResources

File: Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs [hunk 3]
```diff
@@ -1891,26 +2184,32 @@ protected override void OnTemplateApplied()
...
-        }
 
-        /// <inheritdoc/>
-        protected override void UnapplyTemplateCore()
-        {
-            if (this.navigationPanel != null)
+            if (this.dayNameCellStyleCache == null && this.defaultDayNameCellStyle == null)
             {
-                this.navigationPanel.Owner = null;
+                this.defaultDayNameCellStyle = (CalendarCellStyle)RadCalendar.EmbeddedResourceDict["DayNameCellStyle"];
```
Comment (ni4ka7a) [inline Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs]: We can extract the string keys that are repeated in constants

File: Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs [hunk 4]
```diff
@@ -2060,11 +2371,36 @@ private static void OnDisplayModePropertyChanged(DependencyObject target, Depend
...
             calendar.UpdateNavigationHeaderContent();
             calendar.UpdateNavigationPreviousNextButtonsState();
 
+            if (calendar.navigationPanel != null)
+            {
+                if (calendar.displayModeCache == CalendarDisplayMode.MultiDayView)
+                {
+                    if (calendar.navigationPanel.previousButton != null && calendar.navigationPanel.nextButton != null)
+                    {
+                        calendar.navigationPanel.previousButton.Content = "\xE012";
```
Comment (ni4ka7a) [inline Controls/Input/Input.UWP/Calendar/View/RadCalendar.cs]: the same button symbols (**"\xE012"** for example) are used in CalendarNavigationControl - extract a constant so we can easily change them

File: Controls/Core.UWP/Common/WeakCollectionChangedListener.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            var incc = source as INotifyCollectionChanged;
+            if (incc != null)
+            {
+                return new WeakCollectionChangedListener(incc, listener);
+            }
+            return null;
+        }
+
+        internal void Disconnect()
```
[the remark this answers is not in the document]
Comment (APopatanasov) [reply Controls/Core.UWP/Common/WeakCollectionChangedListener.cs]: 👍

File: Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        /// <param name="value">The value produced by the binding source.</param>
+        /// <param name="targetType">The type of the binding target property.</param>
+        /// <param name="parameter">The converter parameter to use.</param>
+        /// <param name="language">The language to use in the converter.</param>
+        /// <returns>
+        /// A converted value. If the method returns null, the valid null value is used.
+        /// </returns>
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            bool flag = false;
```
[the remark this answers is not in the document]
Comment (APopatanasov) [reply Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs]: Will be deleted - it is no longer needed.

File: Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+namespace Telerik.Core
+{
+    /// <summary>
+    /// Represents the converter that converts Boolean values to and from Visibility enumeration values.
+    /// InvertedBooleanToVisibilityConverter converts "true" to Visibility.Collapsed and "false" to Visibility.Visible.
+    /// </summary>
+    public class InvertedBooleanToVisibilityConverter : IValueConverter
+    {
+        /// <summary>
+        /// Converts a value.
```
[the remark this answers is not in the document]
Comment (APopatanasov) [reply Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs]: Will be deleted - it is no longer needed.

File: Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            else if (value is bool?)
+            {
+                bool? nullable = (bool?)value;
+                flag = nullable.Value;
+            }
+            return flag ? Visibility.Collapsed : Visibility.Visible;
+        }
+
+        /// <summary>
+        /// Converts a value.
```
[the remark this answers is not in the document]
Comment (APopatanasov) [reply Controls/Core.UWP/Converters/InvertedBooleanToVisibilityConverter.cs]: Will be deleted - it is no longer needed.

File: Controls/Input/Input.UWP/Calendar/Model/CalendarMultiDayViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,666 @@
...
+
+                this.ArrangeIntersectedAppointments();
+            }
+        }
+
+        private void ArrangeIntersectedAppointments()
+        {
+            foreach (var appointmentInfo in this.appointmentInfos)
+            {
+                if (!appointmentInfo.isIntersected)
```
[the remark this answers is not in the document]
Comment (APopatanasov) [reply Controls/Input/Input.UWP/Calendar/Model/CalendarMultiDayViewModel.cs]: Will be renamed.

**Pull request merged**